adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -235,7 +235,23 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     }
   }
 
-  def loadBam(filePath: String): RDD[AlignmentRecord] = {
+  /**
+   * Loads a SAM/BAM file.
+   *
+   * This reads the sequence and record group dictionaries from the SAM/BAM file
+   * header. SAMRecords are read from the file and converted to the
+   * AlignmentRecord schema.
+   *
+   * @param filePath Path to the file on disk.
+   *
+   * @return Returns a Tuple3 containing (an RDD of reads, the sequence
+   *   dictionary representing the contigs these reads are aligned to
+   *   if the reads are aligned, the record group dictionary for the reads
+   *   if one is available).
+   *
+   * @see loadAlignments
+   */
+  def loadBam(filePath: String): (RDD[AlignmentRecord], SequenceDictionary, RecordGroupDictionary) = {
```

Comment:
> Evidenced by the number of places ._1 had to be added, this is a rather breaking change, even though we aren't committed to binary compatibility at this point.

IIRC, most of the places `._1` had to be added were in tests. I did add it to a few of the CLI commands, but I think that's because most of the commands that we have aren't _transformations_. If you're transforming the data, then you frequently need the sequence/record group dictionaries. If you're just running an aggregation or you're viewing a file, then you may not need the dictionaries.

> My concern though is more from the story telling point -- the main abstraction in Spark is the RDD and ADAM supports loading those from native bioinformatics file formats, e.g. you load a BAM file and get an RDD of AlignmentRecords. Except now that's not the method signature because there's some other stuff too.

I don't think it changes the story really. The sequence/record group dictionaries are just secondary tables of metadata that you frequently want to go with your reads. We could represent them as RDDs if we wanted to, but there's not really a good reason to do that because they're very small (for a single sample) and you're typically going to want to manipulate them on the driver (for sequence dictionaries) or do broadcast joins (with record group dictionaries).